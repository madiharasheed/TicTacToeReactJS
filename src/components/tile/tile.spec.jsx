import React from "react";
import { shallow } from "enzyme";

import Tile from "./tile";
/*Handler Function*/
describe("Tile", () => {
  it("should render Tile component", () => {
    const tileWrapper = shallow(<Tile />);
    expect(tileWrapper).toMatchSnapshot();
  });

  it("when Tile is clicked a handler funciton should be called", () => {
    const mockCallBack = jest.fn();
    const tileWrapper = shallow(<Tile onClick={mockCallBack} />);
    tileWrapper.find("button.tile").simulate("click");
    expect(mockCallBack).toHaveBeenCalled();
  });
});
