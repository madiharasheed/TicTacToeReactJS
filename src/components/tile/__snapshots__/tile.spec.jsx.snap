// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render Tile component 1`] = `
<button
  className="tile undefined"
/>
`;